src/platform/packages/shared/kbn-esql-language/src/commands/definitions/utils/autocomplete/map_expression.ts

```diff
@@ -42,83 +50,126 @@ export type MapParameters = Record<string, MapParameterValues>;
  */
 export function getCommandMapExpressionSuggestions(
   innerText: string,
-  availableParameters: MapParameters
+  availableParameters: MapParameters,
+  includePlaceholder = false
 ): ISuggestionItem[] {
   const finalWord = findFinalWord(innerText);
 
-  // Check if we're inside a nested map by counting braces
-  const openBraces = (innerText.match(/\{/g) || []).length;
-  const closeBraces = (innerText.match(/\}/g) || []).length;
-  const nestingLevel = openBraces - closeBraces;
-
   // Return no suggestions if we're inside a nested map (nesting level > 1)
-  if (nestingLevel > 1) {
+  if (getMapNestingLevel(innerText) > 1) {
     return [];
   }
-
   // Suggest a parameter entry after { or after a comma or when opening quotes after those
   if (/{\s*"?$/i.test(innerText) || /,\s*"?$/.test(innerText)) {
-    const usedParams = new Set<string>();
-    const regex = /"([^"]+)"\s*:/g;
-    let match;
-    while ((match = regex.exec(innerText)) !== null) {
-      usedParams.add(match[1]);
-    }
+    const usedParams = new Set([...innerText.matchAll(/"([^"]+)"\s*:/g)].map(([, name]) => name));
```

Comment:
Can you add it in one constant to be more readable? (also to have consistency) I know it was like that but can we also add it in a constant as the DOUBLE_QUOTE one?